### PR TITLE
add iterator for unique nodes of spectral element space

### PR DESCRIPTION
**Pull request by simonbyrne**
First part of #481.

### CODE REVIEW

File: lib/ClimaCoreTempestRemap/src/ClimaCoreTempestRemap.jl [hunk 1]
```diff
@@ -150,4 +150,30 @@ function write_exodus(
 end
 
 
+function rll_mesh(filename::String; nlat = 90, nlong = round(Int, nlat * 1.6))
```
Comment (jakebolewski) [inline lib/ClimaCoreTempestRemap/src/ClimaCoreTempestRemap.jl:153]: return `filename` / `outfile` on success?

File: src/Spaces/spectralelement.jl [hunk 2]
```diff
@@ -349,3 +349,100 @@ function slab(space::AbstractSpectralElementSpace, v, h)
...
+    Nq = Quadratures.degrees_of_freedom(space.quadrature_style)
+
+    nelem = Topologies.nlocalelems(topology)
+    nvert = length(Topologies.vertices(topology))
+    nface_interior = length(Topologies.interior_faces(topology))
+    if isempty(Topologies.boundary_tags(topology))
+        nface_boundary = 0
+    else
+        nface_boundary = sum(Topologies.boundary_tags(topology)) do tag
+            length(Topologiesboundary_faces(topology, tag))
```
Comment (jakebolewski) [inline src/Spaces/spectralelement.jl:380]:
```suggestion
            length(Topologies.boundary_faces(topology, tag))
```

**Review by jakebolewski — APPROVED** [review]

File: lib/ClimaCoreTempestRemap/src/ClimaCoreTempestRemap.jl (hunk 1 above)
Comment (simonbyrne) [reply lib/ClimaCoreTempestRemap/src/ClimaCoreTempestRemap.jl:153]: this was supposed to be a separate PR, I'll remove it.

**Pull request merged**